test/integration/ModelPhotoShootTest.hh

```diff
@@ -0,0 +1,291 @@
+          getline( iss, word, ' ' );
+          tr_z = std::stof(word);
+          this->translation = {tr_x, tr_y, tr_z};
+          break;
+        }
+        else
+        {
+          if (word == "Scaling:")
+          {
+            getline( iss, word, ' ' );
+            this->scaling = std::stod(word);
+            break;
+          }
+          else
+          {
+            std::string jointName = line.substr(0, line.find(": "));
+            std::string jointPose = line.substr(line.find(": ")+2);
+            jointPositions[jointName] = std::stod(jointPose);
+          }
+        }
+      }
+    }
+    poseFile.close();
+    EXPECT_EQ(remove(poseFilePath.c_str()), 0);
+  }
+
+  /// \brief Tests the Model Photo Shoot plugin with a given sdf world.
+  /// \param[in] _sdfWorld SDF World to use for the test.
+  protected: void ModelPhotoShootTestCmd(const std::string _sdfWorld)
+  {
+    // First run of the server generating images through the plugin.
+    TestFixture fixture(common::joinPaths(std::string(PROJECT_SOURCE_PATH),
+        _sdfWorld));
+    fixture.Server()->SetUpdatePeriod(1ns);
+
+    common::ConnectionPtr postRenderConn;
+    fixture.OnConfigure([&](
+      const Entity &,
+      const std::shared_ptr<const sdf::Element> &,
+      EntityComponentManager &,
+      EventManager &_eventMgr)
+    {
+      postRenderConn = _eventMgr.Connect<gazebo::events::PostRender>(
+            std::bind(&ModelPhotoShootTest::OnPostRender, this));
+    }).Finalize();
+
+    fixture.Server()->Run(true, 50, false);
+    this->LoadPoseValues();
+
+    fixture.OnPreUpdate([&](const gazebo::UpdateInfo &,
+                            gazebo::EntityComponentManager &_ecm)
+    {
+      if(!jointPositions.empty() && this->checkRandomJoints)
+      {
+        _ecm.Each<components::Model>(
+        [&](const ignition::gazebo::Entity &_entity,
+            const components::Model *) -> bool
+        {
+          auto modelName = _ecm.Component<components::Name>(_entity);
+          if (modelName->Data() == "r2")
+          {
+            this->model = std::make_shared<ignition::gazebo::Model>(_entity);
+          }
+          return true;
+        });
+        std::vector<gazebo::Entity> joints = this->model->Joints(_ecm);
+        for (const auto &joint : joints)
+        {
+          auto jointNameComp = _ecm.Component<components::Name>(joint);
+          std::map<std::string, double>::iterator it =
+              jointPositions.find(jointNameComp->Data());
+          if(it != jointPositions.end())
+          {
+            auto jointType = _ecm.Component<components::JointType>
+                (joint)->Data();
+            ASSERT_TRUE(jointType == sdf::JointType::REVOLUTE  ||
+                  jointType == sdf::JointType::PRISMATIC);
+            auto jointAxis = _ecm.Component<components::JointAxis>(joint);
+            ASSERT_GE(it->second,jointAxis->Data().Lower());
+            ASSERT_LE(it->second,jointAxis->Data().Upper());
```

Comment:
I wanted to add a check of the correct position of the `joint` but the entity doesn't seem to contain the `JointPosition` component. @azeey any other suggestion on how to get the position so I can check here?

Reply:
The `JointPosition` component is not available by default. You have to request it by first creating the component on the entity. You then run at least one step and the physics system will populate the component for you. Here's an example: https://github.com/ignitionrobotics/ign-gazebo/blob/2eefd53a946c4657bde4d375172499198f5df970/test/integration/joint_position_controller_system.cc#L78-L83